engine/timer.cpp

```diff
@@ -0,0 +1,50 @@
+#include "timer.h"
+
+#include <spdlog/spdlog.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <memory>
+
+namespace pong
+{
+
+static constexpr int MS_TO_US = 1000;
+static constexpr int S_TO_US = 1000000;
```

Comment:
I would recommend using std::chrono instead of things like this. I think even the frameTime can be changed to some version of a chrono something

Reply:
I'm not sure what you mean by using std::chrono. Is it just a different type? Instead of `int MS_TO_US` it would be `std::chrono::some_type MS_TO_US`? 
And with frameTime, as long as it is easy to use with any calculation (like being multiplied to a vector), I could change that too. `SetPosition(GetPosition() + mVelocity * Timer::frameTime);`

Reply:
Something like this:

```
    // Step 1: Define time in seconds
    std::chrono::seconds sec(5);

    // Step 2: Convert to milliseconds
    std::chrono::milliseconds millis = std::chrono::duration_cast<std::chrono::milliseconds>(sec);

    // Step 3: Output the milliseconds
    std::cout << "Milliseconds: " << millis.count() << std::endl;
```

By keeping all of your time in `chrono` you reduce the chance of messing up some conversion. Note there are no raw integers here, the library abstracts that away.